core/src/main/scala/chisel3/internal/MonoConnect.scala

```diff
@@ -602,6 +596,16 @@ private[chisel3] object checkConnect {
     // Not quite sure where left and right are compared to current module
     // so just error out
     else throw UnknownRelationException
+
+    checkBlockVisibility(sink) match {
+      case Some(blockInfo) => throw SinkEscapedBlockScopeException(sink, blockInfo)
+      case None            => ()
+    }
+
+    checkBlockVisibility(source) match {
+      case Some(blockInfo) => throw SourceEscapedBlockScopeException(source, blockInfo)
+      case None            => ()
+    }
   }
```

Comment:
I don't have a problem with this, but just curious why you moved this below the Module scoping checks?

Reply:
Ahh I see in the commit messages:
```
Report error involving general access or module-level access
before more narrowly diagnosing problem with block scope.

ConnectSpec.scala is fixed with this, but seems generally reasonable.
```